core/src/main/scala/com/softwaremill/diffx/IgnoreMacro.scala

```diff
@@ -2,10 +2,24 @@ package com.softwaremill.diffx
 
 import scala.annotation.tailrec
 import scala.reflect.macros.blackbox
+import scala.reflect.macros.whitebox
 
 object IgnoreMacro {
   private val ShapeInfo = "Path must have shape: _.field1.field2.each.field3.(...)"
 
+  def derivedIgnoreMacro[T: c.WeakTypeTag, U: c.WeakTypeTag](
+      c: whitebox.Context
```

Comment:
why do we need a whitebox macro here? this might make it difficult to port to scala3

Reply:
I haven't really paid much attention to the typ of macro used here, just copy pasted it from the other place. Of course you are right, I will check whether it is possible to change it to black box.